### PR TITLE
Implementation of the observer pattern for a driver change event

### PR DESCRIPTION
**Pull request by askalsky**
Arkadiusz Skalski 218169
Ksawery Szwankowski 209479

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/features/DriverFeature.java [hunk 1]
```diff
@@ -33,6 +34,7 @@ public static void setupDriverPlugin(Application application) {
 	public static void addDriver(String name, Job2dDriver driver) {
 		SelectDriverMenuOptionListener listener = new SelectDriverMenuOptionListener(driver, driverManager);
 		app.addComponentMenuElement(DriverFeature.class, name, listener);
+		driverManager.getPublisher().addSubscriber(new DriverLabelObserver());
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/DriverFeature.java]: The observer should be subscribed only once. Now it is done each time a driver is added application.

**Review by kowallus — CHANGES_REQUESTED** [review]
Only one thing is done incorrectly.

File: src/main/java/edu/kis/powp/jobs2d/features/DriverFeature.java (hunk 1 above)
Comment (askalsky) [reply src/main/java/edu/kis/powp/jobs2d/features/DriverFeature.java]: @kowallus I think the current solution should solve the problem

**Comment by askalsky** [comment]
@kowallus We have improved the solution

**Pull request merged**